### DISCUSSION TITLE
Avatar not displaying on commits

### DISCUSSION THREAD

**Discussion by thelaurafranklin** [community/community#21643] [New to GitHub]
<p>I hope this is the right place to ask my question…<br>
I recently changed my GitHub username.  Dangerous, I know, but I really needed to for consistency.  It seemed to work fine.  I updated my .gitconfig file with my new username and primary email (yeah, changed that, too).  I confirmed in the terminal that it was recognizing the changes.</p>
<p>I should mention that I am using Mac OS High Sierra and Atom as my IDE.  When I made an edit to a project file in Atom and then opened the Git pane, I noticed my avatar wasn’t displaying.  Instead it is the default one.  When I pushed the commit to GitHub, I noticed the avatar displayed next to commits was the octacat.  The name associated with the commits has changed, however, so it appears it is recognizing that change, but not displaying the avatar.</p>
<p>I did change my avatar as part of this process, and previous commits are displaying the new avatar instead of the old on both Atom and GitHub.  It is just all new commits that are being affected.</p>
<p>Any suggestions?</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="thelaurafranklin" data-post="1" data-topic="127617">
<div class="title">
<div class="quote-controls"></div>
 thelaurafranklin:</div>
<blockquote>
<p>I updated my .gitconfig file with my new username and primary email (yeah, changed that, too).</p>
</blockquote>
</aside>
<p>Did you <a href="https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/adding-an-email-address-to-your-github-account">add your new primary email to your Github account</a>, too? That’s what Github uses to detect who authored commits. If it doesn’t recognize the email address it’ll use the default avatar and the name in the commit.</p>

**Reply by okabeparry** [reply]
#1

**Reply by okabeparry** [reply]
#2

**Comment by thelaurafranklin** [comment]
<p>Yes, I changed my username, primary email, and profile picture first.  Then I updated the .gitconfig with the username and email.</p>
<p>To be clear:  The only part that isn’t working is the avatar the normally appears next to commits.  It is displaying default avatars in both Atom and GitHub.  The username is displaying next to commits in GitHub and it is updating the contributions with new commits.</p>

**Comment by thelaurafranklin** [comment]
<p>Just decided to remove my profile pic and see what happens.  The pic changed to one of those patterned ones and all of the older commits changed to that avatar.  The commits since the change, however, continue to have the octacat avatar.</p>

**Comment by airtower-luna** [comment]
<p>I took a quick look at your Github profile, and I can see that Github clearly doesn’t recognize the newer commits as yours. Yes, the listed name is the same as your username (presumably because it’s the author name in the commit), but it’s not linked to your profile like the old ones are.</p>
<p>What I can’t say is why that is. If the email is correct (exactly the same in your commits and in your profile) it’d have to be some Github-internal issue. Maybe caching regarding which mail addresses are on which account? But I can only speculate there.</p>

**Comment by thelaurafranklin** [comment]
<p>If that is the case, then why are my new commits showing up as contributions??<br>
I have submitted a ticket to GitHub in case it is an internal issue.  Thanks for your help!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="thelaurafranklin" data-post="6" data-topic="127617">
<div class="title">
<div class="quote-controls"></div>
 thelaurafranklin:</div>
<blockquote>
<p>If that is the case, then why are my new commits showing up as contributions??</p>
</blockquote>
</aside>
<p>I don’t think they are: Your public profile shows one contribution for today, the creation of a repository. The commits you pushed to that repository do not show up so far, though that could also be a matter of caching.</p>

**Comment by thelaurafranklin** [comment]
<p>Aha!  You are right.  It shows that I made the new repository, but I just made test commits on ‘test’ and ‘kitten-clicker’ and neither of those are showing up.  hmmm…</p>
<p>Could you do me a favor and do a pull request on ‘test’. just add a comment or something.  I want to confirm that it works fine for other people.</p>

**Comment by thelaurafranklin** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a><br>
Wow.  I am so sorry for wasting your time.  After you pointed out that the commits weren’t showing up in my contributions, I started investigating and found out that if you add .patch to a commit it displays information about the commit.  Somehow it wasn’t until I viewed my email address there that I realized I was missing a letter in my last name.  Fixed it and now it is working properly.  Thanks for your help.<br>
<img src="https://user-images.githubusercontent.com/31911635/181065965-9885e15a-db77-4f95-9707-2e84bfbb0137.gif" alt="" data-base62-sha1="uNSgPTs78qX6MhYSw91zNpLGErl" width="640" height="480"></p>

**Comment by airtower-luna** [comment]
<p>Thank you for the update, I’m glad you figured it out! That’s about what I suspected at first. :grin:</p>

**Comment by CxrlosKenobi** [comment]
<p>I would like to add that I just had the same problem with my commits, and after reading this I solved it checking my git credentials on my computer with <code>git config --list</code>, and there I notice that my user.email was incorrect, so I changed it with <code>git config --global user.email 'my email'</code>. And that solved at all my struggle.</p>
<p>I hope this can help to someone that had the same problem.</p>

**Reply by mjarros** [reply]
I had the same problem with my user.email being incorrect. After correcting, the problem was solved. Thanks for sharing the advice. =D

**Comment by Pallavi-PH** [comment]
<p>Yes, this happens when the username or email provided in git config is incorrect from the one registered in GitHub</p>

**Comment by alessandroamormino** [comment]
Hi everyone, 
i have the same issue here, no profile pic is showing in my commits (I use VS Code). 
I've checked my git config and both the username and email variable are set correctly with my github account and my mail is the primary one.
I also tried to clear cache in the gitlents extension in VSC thinking that it may help but nothing changes. 
The commits are working because i work with this account and in my company it works just fine. 
t
Any suggestions? 
Thanks!

**Reply by shivam01anand** [reply]
same issue. found a soln? @alessandroamormino

**Reply by alessandroamormino** [reply]
Yeah, i've cleaned the gitlents cache and then remove/add the github env variable in my global PATH even if they were already fine. After a couple of days the issue solved itself

**Comment by angel-zeitoune** [comment]
In the GitLens commit list, it is necessary to update the image at [https://gravatar.com/](https://gravatar.com/) associated with your email

**Comment by tahmidefaz** [comment]
Setting the following global git configs fixed it for me:
```
git config --global user.name '<your_github_username>'
git config --global user.email '<your_github_email_address>'
```
